diff -u -r ./CMakeLists.txt ../chai-2.2.2-patched/CMakeLists.txt
--- ./CMakeLists.txt	2020-10-12 12:36:21.000000000 -0700
+++ ../chai-2.2.2-patched/CMakeLists.txt	2021-07-13 20:53:51.419659781 -0700
@@ -6,6 +6,7 @@
 ##############################################################################
 cmake_policy(SET CMP0057 NEW)
 cmake_policy(SET CMP0048 NEW)
+cmake_policy(SET CMP0025 NEW)
 
 project(Chai LANGUAGES CXX VERSION 2.2.2)
 
