diff --git a/tpl/camp/include/camp/resource/host.hpp b/tpl/camp/include/camp/resource/host.hpp
index c765175..4c12f98 100644
--- a/tpl/camp/include/camp/resource/host.hpp
+++ b/tpl/camp/include/camp/resource/host.hpp
@@ -11,6 +11,8 @@ http://github.com/llnl/camp
 #ifndef __CAMP_HOST_HPP
 #define __CAMP_HOST_HPP
 
+#include <cstdlib>
+
 #include "camp/resource/event.hpp"
 #include "camp/resource/platform.hpp"
 
@@ -54,7 +56,7 @@ namespace resources
       template <typename T>
       T *allocate(size_t n, MemoryAccess = MemoryAccess::Device)
       {
-        return (T *)malloc(sizeof(T) * n);
+        return (T *)std::malloc(sizeof(T) * n);
       }
       void *calloc(size_t size, MemoryAccess = MemoryAccess::Device)
       {
@@ -62,7 +64,7 @@ namespace resources
         this->memset(p, 0, size);
         return p;
       }
-      void deallocate(void *p, MemoryAccess = MemoryAccess::Device) { free(p); }
+      void deallocate(void *p, MemoryAccess = MemoryAccess::Device) { std::free(p); }
       void memcpy(void *dst, const void *src, size_t size) { std::memcpy(dst, src, size); }
       void memset(void *p, int val, size_t size) { std::memset(p, val, size); }
     };
