diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6d9b073a..a15b54c3 100644
--- a/src/tpl/umpire/CMakeLists.txt
+++ b/src/tpl/umpire/CMakeLists.txt
@@ -93,10 +93,12 @@ if (NOT BLT_LOADED)
   include(${BLT_SOURCE_DIR}/SetupBLT.cmake)
 endif()
 
-blt_git_hashcode (HASHCODE umpire_sha1
-                  RETURN_CODE rc
-                  SOURCE_DIR ${PROJECT_SOURCE_DIR})
-set (UMPIRE_VERSION_RC ${umpire_sha1})
+if (Git_FOUND)
+  blt_git_hashcode (HASHCODE umpire_sha1
+                    RETURN_CODE rc
+                    SOURCE_DIR ${PROJECT_SOURCE_DIR})
+  set (UMPIRE_VERSION_RC ${umpire_sha1})
+endif()
 
 include(cmake/SetupCMakeBasics.cmake)
 include(cmake/SetupCompilerFlags.cmake)
