diff --git a/src/parcsr_mv/par_csr_matrix.c b/src/parcsr_mv/par_csr_matrix.c
--- a/src/parcsr_mv/par_csr_matrix.c
+++ b/src/parcsr_mv/par_csr_matrix.c
@@ -2458,1 +2458,1 @@
-   HYPRE_Int                  contact_send_buf[1];
+   HYPRE_BigInt               contact_send_buf[1];
@@ -2504,1 +2504,1 @@
-      contact_send_buf[0]  = (HYPRE_Int) hypre_ParCSRMatrixLastRowIndex(par_matrix);
+      contact_send_buf[0]  = hypre_ParCSRMatrixLastRowIndex(par_matrix);
@@ -2533,1 +2533,1 @@
-                          contact_send_buf_starts, sizeof(HYPRE_Int),
+                          contact_send_buf_starts, sizeof(HYPRE_BigInt),


